x-pack/plugins/lens/public/xy_visualization/get_legend_action.tsx

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import type { LegendAction, XYChartSeriesIdentifier } from '@elastic/charts';
+import type { LayerArgs } from './types';
+import type { LensMultiTable, LensFilterEvent } from '../types';
+import { LegendActionPopover } from '../shared_components';
+
+export const getLegendAction = (
+  filteredLayers: LayerArgs[],
+  tables: LensMultiTable['tables'],
+  onFilter: (data: LensFilterEvent['data']) => void
+): LegendAction => ({ series: [xySeries], label }) => {
+  const series = xySeries as XYChartSeriesIdentifier;
+  const layer = filteredLayers.find((l) =>
+    series.seriesKeys.some((key: string | number) => l.accessors.includes(key.toString()))
+  );
+
+  if (!layer || !layer.splitAccessor) {
+    return null;
+  }
+
+  let splitLabel = label;
+  const accessor = layer.splitAccessor;
+  const hasMultipleYaxis = layer.accessors.length > 1;
+  if (hasMultipleYaxis) {
```

Comment:
Why are you not using `xySeries.seriesKeys[0]` for this? Seems like a more stable approach

Reply:
I wanted to avoid the formatter, for example 
![image](https://user-images.githubusercontent.com/17003240/122564764-670f1400-d04e-11eb-92c5-ff5ed3705cdf.png)

I will give it a try though!

Reply:
Right, for the visible label you would have to run it through the formatter again - but you anyway need it in there for the issue with object values I reported below

Reply:
You are absolutely right as always 🙂